@ecomplus/storefront-components/src/js/ProductCard.js

```diff
@@ -192,6 +207,7 @@ export default {
       if (this.product.visible === undefined) {
         this.body.visible = true
       }
+      this.isFavorite = checkFavorite(this.body._id, this.ecomPassport)
```

Comment:
```suggestion
```

ué ? 😅 
nada a ver com o `buy`, isso deveria acontecer no hook `created`, `mounted` ou até direto no data inicial, pra você setar o estado do `isFavorite` no início